File: backend/app/Domain/Doctor/Http/Requests/ResetPasswordRequest.php
<?php

namespace App\Domain\Doctor\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

class ResetPasswordRequest extends FormRequest
{
    public function authorize()
    {
        return true;
    }

    public function rules()
    {
        return [
            'identifier' => 'required|string',
            'code' => 'required|string',
            'password' => 'required|string|min:8|confirmed',
        ];
    }
}

File: mobile_app/respilink_mobile/lib/features/auth/data/models/requests/reset_password_request.dart
class ResetPasswordRequest {
  final String email, otp, password, confirmPassword;

  ResetPasswordRequest({
    required this.email,
    required this.password,
    required this.otp,
    required this.confirmPassword,
  });

  Map<String, dynamic> toJson() {
    return {
      'identifier': email,
      'code': otp,
      'password_confirmation': confirmPassword,
      'password': password,
    };
  }
}
